base/range.jl

```diff
@@ -1480,3 +1480,111 @@ julia> mod(3, 0:2)  # mod(3, 3)
 """
 mod(i::Integer, r::OneTo) = mod1(i, last(r))
 mod(i::Integer, r::AbstractUnitRange{<:Integer}) = mod(i-first(r), length(r)) + first(r)
+
+
+"""
+    LogRange(start, stop, length)
+
+Construct a specialized array whose elements are spaced logarithmically
+between the given endpoints. That is, the ratio of successive elements is
+a constant, calculated from the length.
+
+Like [`LinRange`](@ref), floating-point error may cause it to miss intermediate
+rational values, but the first and last elements should be exactly those provided.
+
+This is similar to `geomspace` in Python. Unlike `PowerRange` in Mathematica,
+you specify the number of elements not the ratio.
+Unlike `logspace` in Python and Matlab, the `start` and `stop` arguments are
+always the first and last elements of the result, not powers applied to some base.
+
+!!! compat "Julia 1.9"
+    This function requires at least Julia 1.9.
+
+# Examples
+```jldoctest
+julia> LogRange(10, 4000, 3)
+3-element LogRange{Float64}:
+ 10.0, 200.0, 4000.0
+
+julia> ans[2] ≈ sqrt(10 * 4000)  # middle element is the geometric mean
+true
+
+julia> LogRange(-62_500, -0.01, 5) |> collect  # endpoints are always exact
+5-element Vector{Float64}:
+ -62500.0
+  -1250.0
+    -25.0
+     -0.49999999999999994
+     -0.01
+
+julia> LogRange(√2, 32, 10)
+10-element LogRange{Float64}:
+ 1.41421, 2.0, 2.82843, 4.0, 5.65685, 8.0, 11.3137, 16.0, 22.6274, 32.0
+
+julia> LogRange(1, 1000, 4) ≈ 10 .^ (0:3)  # was logspace(0,3,4) in Julia < 1.0
+true
+
+julia> LogRange(1, -1+0im, 5) ≈ cis.(LinRange(0, pi, 5))  # complex numbers
+true
+```
+"""
+struct LogRange{T} <: AbstractArray{T,1}
+    start::T
+    stop::T
+    len::Int
+    function LogRange(start::T1, stop::T2, len::Integer) where {T1<:Number, T2<:Number}
+        T = float(promote_type(T1, T2))
+        lo = iszero(start) ? T(NaN) : T(start)
+        hi = iszero(stop) ? T(NaN) : T(stop)
+        if len < 0
+            throw(ArgumentError("LogRange can't have negative length"))
+        elseif len == 0
+            return new{T}(lo, hi, len)
+        elseif len == 1 && start != stop
+            throw(ArgumentError("endpoints of LogRange differ, with length = 1"))
```

Comment:
As above, unless there is a reason to differ, this should be the same as LinRange.
```julia
julia> LinRange(1,2,1)
ERROR: ArgumentError: range(1.0, stop=2.0, length=1): endpoints differ
```
Perhaps:
```suggestion
            throw(ArgumentError("LogRange($start, $stop, $len): endpoints differ"))
```
or if you prefer keeping "with negative length", make that change in LinRange too.

Reply:
That's fine. Was more careful about error cases than about matching the wording.

Note that `LinRange` has lots of truly ancient printing code, some of which might now be done differently. In particular, this one has the obvious mistake of saying `range` meaning it's untouched since Julia 0.4 or something.